src/databricks/labs/blueprint/paths.py

```diff
@@ -448,17 +441,296 @@ def match(self, path_pattern, *, case_sensitive=None):
+        """Return the absolute path of the file or directory in Databricks Workspace."""
+        absolute = self.absolute()
+        if strict and not absolute.exists():
+            msg = f"Path does not exist: {self}"
+            raise FileNotFoundError(msg)
+        return absolute
+
+    def absolute(self: P) -> P:
+        if self.is_absolute():
+            return self
+        return self.with_segments(self.cwd(), self)
+
+    def _prepare_pattern(self, pattern: str | bytes | os.PathLike) -> Sequence[str]:
+        if not pattern:
+            raise ValueError("Glob pattern must not be empty.")
+        parsed_pattern = self.with_segments(pattern)
+        if parsed_pattern.anchor:
+            msg = f"Non-relative patterns are unsupported: {pattern!s}"
+            raise NotImplementedError(msg)
+        pattern_parts = parsed_pattern._path_parts  # pylint: disable=protected-access
+        if ".." in pattern_parts:
+            msg = f"Parent traversal is not supported: {pattern!s}"
+            raise ValueError(msg)
+        if os.fspath(pattern)[-1] == self.parser.sep:
+            pattern_parts = (*pattern_parts, "")
+        return pattern_parts
+
+    def glob(
+        self: P, pattern: str | bytes | os.PathLike, *, case_sensitive: bool | None = None
+    ) -> Generator[P, None, None]:
+        pattern_parts = self._prepare_pattern(pattern)
+        if case_sensitive is None:
+            case_sensitive = True
+        selector = _Selector.parse(pattern_parts, case_sensitive=case_sensitive)
+        yield from selector(self)
+
+    def rglob(
+        self: P, pattern: str | bytes | os.PathLike, *, case_sensitive: bool | None = None
+    ) -> Generator[P, None, None]:
+        pattern_parts = ("**", *self._prepare_pattern(pattern))
+        if case_sensitive is None:
+            case_sensitive = True
+        selector = _Selector.parse(pattern_parts, case_sensitive=case_sensitive)
+        yield from selector(self)
+
+    @abstractmethod
+    def as_uri(self) -> str: ...
+
+    @abstractmethod
+    def as_fuse(self) -> Path: ...
+
+    @abstractmethod
+    def exists(self, *, follow_symlinks: bool = True) -> bool: ...
+
+    @abstractmethod
+    def mkdir(self, mode: int = 0o600, parents: bool = True, exist_ok: bool = True) -> None: ...
+
+    @abstractmethod
+    def rmdir(self, recursive: bool = False) -> None: ...
+
+    @abstractmethod
+    def unlink(self, missing_ok: bool = False) -> None: ...
+
+    @abstractmethod
+    def open(
+        self,
+        mode: str = "r",
+        buffering: int = -1,
+        encoding: str | None = None,
+        errors: str | None = None,
+        newline: str | None = None,
+    ): ...
+
+    @abstractmethod
+    def is_dir(self) -> bool: ...
+
+    @abstractmethod
+    def is_file(self) -> bool: ...
+
+    @abstractmethod
+    def expanduser(self: P) -> P: ...
+
+    @abstractmethod
+    def iterdir(self: P) -> Generator[P, None, None]: ...
```

Comment:
```suggestion
    def iterdir(self: P) -> Iterable[P]: ...
```
i wonder if it could be iterable?

Reply:
I wondered the same, but Mypy was unhappy about that: the problem is that `Iterable` widens the return type from the method and callers might not expect that.